template/extra_runtime.go

```diff
@@ -377,14 +377,20 @@ func (t *Template) tryConvert(value string) interface{} {
 }
 
 // Execute the command (command could be a file, a template or a script) and convert its result as data if possible
-func (t *Template) exec(command string, args ...interface{}) (result interface{}, err error) {
-	if result, err = t.run(command, args...); err == nil {
-		if result == nil {
-			return
-		}
-		result = t.tryConvert(result.(string))
+func (t *Template) exec(command string, args ...interface{}) (interface{}, error) {
+	commandOutput, err := t.run(command, args...)
+	if err != nil || commandOutput == nil {
+		return commandOutput, err
+	}
+
+	var parsedOutput interface{}
+	err = collections.ConvertData(commandOutput.(string), &parsedOutput)
```

Comment:
If I understand correctly by my quick review of it, the `tryConvert` function will try to template so this is why you removed it in favour of the `ConvertData` direct use?

Reply:
Yeah. `tryConvert` ends up calling `t.dataConverter` which is the function that gets called when you use the `data` function in gotemplate code. That function ends up templating stuff.